gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/ClientCredentialsResource.java

```diff
@@ -75,7 +75,7 @@ public Response getAuthenticationToken() {
 
         TokenResponseContext resp = getTokenResponse(context);
         if (resp.responseMap != null) {
-            String passcode = resp.responseMap.passcode;
+            String passcode = (String) resp.responseMap.map.get(PASSCODE);
```

Comment:
Why do you not need to reference the token id similarly?

Reply:
Yeah, this was the source of the initial bug in the implementation. For some reason the tokenid and the passcode are surfaced at the top level but the passcode there is just the passcode portion of what ends up being the client_secret which contains both. I didn't really dig into why there are two levels but the one inside the map is put together and base64 encoded properly for use where the other is just the UUID format which can't be used directly.

Reply:
I appears to be related to the fact that when there is no token state server configured that the passcode field isn't generated since it can't be persisted and it requires server side state. Why the actual passcode UUID is generated and returned in the out response is still not clear.